include/revng-c/mlir/Dialect/Clift/IR/CliftAttributes.h

```diff
@@ -42,40 +42,40 @@ class StructType
 public:
   using Base::Base;
 
-  static StructType get(MLIRContext *ctx, uint64_t ID);
+  static StructType
+  get(MLIRContext *Context, uint64_t ID, llvm::StringRef Name, uint64_t Size);
 
   static StructType
   getChecked(llvm::function_ref<InFlightDiagnostic()> EmitError,
-             MLIRContext *ctx,
-             uint64_t ID);
+             MLIRContext *Context,
+             uint64_t ID,
+             llvm::StringRef Name,
```

Comment:
:thinking: why do we need the name now?

Also, even if we really need it, being it optional I'd leave it as last argument (which I think also leads to be able to drop this commit ae216398ab4788bc44567adb8febb1b76916be0c )

Reply:
Same for the previous `get` method.

Reply:
We don't *need* the name yet. There's two ways to look at the split:
Should we take up front only the things we really need? Or should we take up front everything that we can and only specify later the things that can only be specified later? I went with the latter, but I think either is fine.
I made it the last argument mainly for symmetry with `UnionType`. This way they have a common prefix. This was mainly important for the storage type template, but I kept it consistent for the attribute types.

Reply:
I think we should take upfront only what we really need.
Do you have any compelling reason to do the opposite?

If we do like I propose, AFAICT then both `UnionType` and `StructType` have a common prefix, that is needed by both, and it's only the ID. Makes sense?

Reply:
I changed this in the `Review changes 2 (to be squashed)` commit. Once you're happy with the changes, I'll squash it into the previous commits.

Reply:
Thanks!